execution/limit.go

```diff
@@ -68,19 +71,32 @@ func (node *LimitedStream) Close() error {
 }
 
 func (node *LimitedStream) Next(ctx context.Context) (*Record, error) {
-	if node.limit > 0 {
+	tx := storage.GetStateTransactionFromContext(ctx)
+	limitState := storage.NewValueState(tx.WithPrefix(node.streamID.AsPrefix()).WithPrefix(limitPrefix))
+
+	var limit octosql.Value
+	err := limitState.Get(&limit)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't get limit state")
+	}
+
+	if limit.AsInt() > 0 {
 		record, err := node.rs.Next(ctx)
 		if err != nil {
 			if err == ErrEndOfStream {
-				node.limit = 0
+				limit = octosql.MakeInt(0)
```

Comment:
this variable is not used because it's followed by early return (?)

Reply:
It will be used in case someone tries to get a record again.

Reply:
Ah, I didn't save it.